unittests/allplatformstests.py

```diff
@@ -2049,7 +2049,7 @@ def test_subproject_promotion(self):
 
     def test_subproject_promotion_wrap(self):
         testdir = os.path.join(self.unit_test_dir, '43 promote wrap')
-        workdir = os.path.join(self.builddir, 'work')
+        workdir = os.path.join(self.builddir, '..', 'work')
```

Comment:
This cannot use the parent directory of a tmpdir, it needs to create its own tmpdir. As is, this leaves a directory "work" behind in the root of the meson git repository.

You can do this by invoking `self.new_builddir()` rather than changing the value of `workdir`.

Reply:
My PR adds the line
```
builddir = os.path.join(newdir, 'build')
```
so, `os.path.join(self.builddir, '..', 'work')` is `meson/tmp9buhvdhp/build/../work` which is `meson/tmp9buhvdhp/work`

No folders are left behind.

Reply:
Sorry, you're correct. Instead, it leaves `tmp9buhvdhp/work/` behind. We only clean up self.builddir, we do not clean up self.builddir/../

TBH it's not obvious to me that refactoring new_builddir instead of just calling it again is worth the fuss of getting this right.